src/_gettsim/functions/preprocessor.py

```diff
@@ -127,7 +126,7 @@ def load_and_check_functions(
 
 
 def _create_derived_functions(
```

Comment:
I think I'd prefer being stricter about `PolicyFunction`s here -- just replace `Callable` by that in all code underneath this call. Makes it clearer what is the purpose of things instead of relying on the sanitizing in `PolicyFunction.__init__`.

Also, instead of returning a tuple with three `dict[str, PolicyFunction]`s, let's just return a flat dictionary with all three tuple items. We never need a distinction of those again.

Reply:
Update after talking on the phone, just for the public record: This will become something like `PolicyFunction | DerivedFunction`

Reply:
Union types can be hard to use, since they only allow access to operations that are defined on all its elements. Otherwise, `isinstance` checks must be added, which can affect many locations in the code.

Generally, I don't see a major benefit of adding a completely separate class for `DerivedFunction` instead of using `PolicyFunction` for them, too: The fields `function`, `module_name`, and `function_name` are also required for derived functions. `start_date` and `end_date` are useful information. Rounding and vectorization are questionable, but could still enable some use-cases.

Tracking information could become part of `PolicyFunction` and be `None` for non-derived functions. If `DerivedFunction` was a subclass of `PolicyFunction`, we would first have to add an `isinstance` check, before we can access which function we derived from, which is redundant.

Reply:
Thanks! From my perspective, it is mostly about communicating intent and source (written code vs. automatically generated). Granted, some of the `PolicyFunction`s are not about (public) policy at all, either, but for sure none of the `DerivedFunction`s is. So the name would be off.

I would be fine with just making `DerivedFunction` have the same signature as `PolicyFunction` and fill the fields with default values. 

This said, I think that `start_date` and `end_date` should be irrelevant at the point where `DerivedFunction`s come into play *(or am I missing something? Only happens once we condition on a particular point in time, no?)* so not accessing them might even be a helpful sanity check.

Reply:
> This said, I think that start_date and end_date should be irrelevant at the point where DerivedFunctions come into play

Yes, they would just be used for introspection.

Reply:
@hmgaudecker The class for derived functions is now added, too.